### ISSUE TITLE
Why is the credentials path required when using postgresql or mysql as the data provider?

### ISSUE THREAD

**Issue by djmattyg007** [drakkan/sftpgo#140]
I'm getting a permissions error when SFTPGo tries to create the `credentials` directory automatically, even though I'm using a data provider that should not require any credentials to be written to disk. It isn't clear why this is necessary.

**Comment by sergiz** [comment]
Credentials path is not used by data provider, but by sotrage backend (google cloud currently). But yeah, I agree it should be ignored, if a backend that actually requires credentials path is not used.

**Comment by drakkan** [comment]
Hi, 

the code assume that the credentials directory is there and it is writable. As @sergiz explained this directory will store GCS credentials and it will be used if you create a GCS user via REST API or Web admin. 

Basically you need to create this directory in advance or let sftpgo create it for you. If you don't create GCS users it will remain empty.

**Comment by djmattyg007** [comment]
Thanks. It seems odd that it's stored alongside other settings that are actually related to the database.

**Comment by drakkan** [comment]
Hi,

for S3, credentials are a simple key/secret pair and they are stored (encrypted) directly inside the data provider. For GCS I initially evaluated to store credentials in the data provider too but these files have a discrete size, they are useful only when the user login. Storing them inside the data provider will increase its size with no real advantage, and sending them inside the get user/s HTTP response body could be a privacy violation too. So I decided to store them, unmodified, in the `credentials` folder. They are read when a GCS user try to login or when you request a users dump. A users dump is stored locally and it is not sent over the network.

So basically they are a filesytem extension for the data provider and they are dumped/restored as the others data provider fields. This is the reason because they are configured inside the dataprovider section.

In the [docs](https://github.com/drakkan/sftpgo/blob/master/docs/full-configuration.md) I documented this confiuration key in this way:

- `credentials_path`, string. It defines the directory for storing user provided credential files such as Google Cloud Storage credentials. This can be an absolute path or a path relative to the config dir

do you think the docs can be improved or do you have other suggestions? Thanks!

**Comment by djmattyg007** [comment]
> do you think the docs can be improved or do you have other suggestions? Thanks!

I think there would be less confusion about it if the credentials folder was by default stored in `/var/lib` (which should be writable by the `sftpgo` user) rather than `/etc/sftpgo` (which IMO should only be writable by the `root` user).

**Comment by drakkan** [comment]
Hi,

Arch Linux package use [/var/lib](https://aur.archlinux.org/cgit/aur.git/tree/sftpgo.json?h=sftpgo#n52) and Linux specific documentation suggests [/var/lib](https://github.com/drakkan/sftpgo/blob/master/docs/service.md) for credentials too. 

The default configuration could not be changed to use /var/lib since this is a Linux specific path that does not work at all on Windows and is suboptimal on macOS and *BSD

**Comment by djmattyg007** [comment]
Is there no ability to customise the default on a per-OS basis?

I think the best way to handle it would be to introduce a new setting named something like "runtime path" that (for Linux) would point to `/var/lib` by default (mimicking the "configuration path", which for Linux points to `/etc` by default). `credentials_path` would then be relative to the runtime path setting by default.

**Comment by drakkan** [comment]
I'm a bit unsure here, there are too many options already, I want to limit the number of configuration options to avoid to confuse new users. This could be managed by packager as I did in AUR PKGBUILD. I'll leave the ticket open to think about this request a bit more, thanks

**Comment by drakkan** [comment]
Hi,

I think this must be done by packagers, we already have a lot of options and this doesn't seem essential, sorry
